feat: add schema name caching and WebVH support

PatStLouis opened the pull request:
# Summary of Changes

This improves credential display by caching the schema name at offer time and using it when OCA overlays don't provide a specific name or only return generic names like 'Credential'. Also enables support for WebVH-based credentials.

- Add getSchemaName() utility to retrieve cached schema name from credential metadata
- Store schema name in AnonCredsCredentialMetadataKey when credentials are offered
- Use cached schema name as fallback when OCA overlay doesn't provide meaningful name
- Add WebVhAnonCredsRegistry and WebvhDidResolver for did:webvh credential support
- Update credential cards and details views to use cached schema name
- Optimize history logging to avoid unnecessary async calls when cached name exists
- Add tests for schema name retrieval and caching

# Testing Instructions

Issue and present a credential without OCA. The credential name should use the schema name value.

# Acceptance Criteria

Schema name is used for credential label when no oca bundle exists.

# Screenshots, videos, or gifs

N/A

# Breaking change guide

If users where relying on cred def tags for display, this will no longer work.

# Related Issues

#1672

# Pull Request Checklist

Tick all boxes below to demonstrate that you have completed the respective task. If the item does not apply to your this PR **check it anyway** to make it apparent that there's nothing to do.

- [X] All commits contain a DCO `Signed-off-by` line (we use the [DCO GitHub app](https://github.com/apps/dco) to enforce this)
- [X] If applicable, screenshots, gifs, or video are included for UI changes
- [X] If applicable, breaking changes are described above along with how to address them
- [X] If applicable, added [changeset(s)](https://github.com/changesets/changesets)
- [X] Added sufficient [tests](../packages/core/__tests__/) so that overall code coverage is not reduced

If you have _any_ questions to _any_ of the points above, just **submit and ask**! This checklist is here to _help_ you, not to deter you from contributing!

Pro Tip 🤓

- Read our [contribution guide](../CONTRIBUTING.md) at least once; it will save you a few review cycles!
- Your PR will likely not be reviewed until all the above boxes are checked and all automated checks have passed

PatStLouis committed:
feat: Add schema name caching and WebVH support

- Add getSchemaName() utility to retrieve cached schema name from credential metadata
- Store schema name in AnonCredsCredentialMetadataKey when credentials are offered
- Use cached schema name as fallback when OCA overlay doesn't provide meaningful name
- Add WebVhAnonCredsRegistry and WebvhDidResolver for did:webvh credential support
- Update credential cards and details views to use cached schema name
- Optimize history logging to avoid unnecessary async calls when cached name exists
- Add tests for schema name retrieval and caching

This improves credential display by caching the schema name at offer time
and using it when OCA overlays don't provide a specific name or only return
generic names like 'Credential'. Also enables support for WebVH-based credentials.

Signed-off-by: Patrick St-Louis <patrick.st-louis@opsecid.ca>

PatStLouis committed:
Merge branch 'main' into feature/anoncreds-metadata-caching

PatStLouis committed:
Merge branch 'main' into feature/anoncreds-metadata-caching

PatStLouis committed:
fix: add changeset

Signed-off-by: Patrick St-Louis <patrick.st-louis@opsecid.ca>

jleach commented on packages/core/src/components/misc/CredentialCard10.tsx:
```diff
@@ -153,11 +154,33 @@ const CredentialCard10: React.FC<CredentialCard10Props> = ({ credential, style =
+      const cachedSchemaName = getSchemaName(credential)
+      // Check if OCA name is meaningful (not just "Credential" or empty)
+      const ocaName = bundle.metaOverlay?.name
+      const hasMeaningfulOcaName = ocaName && ocaName !== 'Credential' && ocaName.trim() !== ''
```
Could this be a custom use case to your current testing scenario? For example why not "Le justificatif" in French or "Der Ausweis" in German rather than "Credential" in English? Where I'm going here is are we assuming the English word is always used here? I though OCA always had the built in concept of i8n - no?

PatStLouis replied:
'Credential' seemed to be a hard value coded in my experience

jleach commented on packages/core/src/components/misc/CredentialCard11.tsx:
```diff
@@ -235,11 +236,33 @@ const CredentialCard11: React.FC<CredentialCard11Props> = ({
+      // Use cached schema name as fallback if no specific overlay name is found
+      const cachedSchemaName = credential ? getSchemaName(credential) : undefined
+      const ocaName = bundle.metaOverlay?.name
+      const hasMeaningfulOcaName = ocaName && ocaName !== 'Credential' && ocaName.trim() !== ''
```
Since "Credential" is used in a few places and may change depending on the native language of the wallet should this go in a constants file?

PatStLouis replied:
Let me review where it was already used in the code, it was already a fall back value. I like the idea of making it a variable DefaultCredentialNameValue or something to make it clearer

PatStLouis replied:
added a global variable

jleach commented on packages/core/src/components/misc/CredentialCard10.tsx:
```diff
@@ -153,11 +154,33 @@ const CredentialCard10: React.FC<CredentialCard10Props> = ({ credential, style =
     }
     bundleResolver.resolveAllBundles(params).then((bundle) => {
+      // Use cached schema name as fallback if no specific overlay name is found
+      const cachedSchemaName = getSchemaName(credential)
```
Below you do `credential ? getSchemaName(credential) : undefined`. Should we do that here also for consistency?

PatStLouis replied:
I will look into it

PatStLouis replied:
implemented a shared function

jleach commented on packages/core/src/screens/CredentialDetails.tsx:
```diff
@@ -172,14 +174,68 @@ const CredentialDetails: React.FC<CredentialDetailsProps> = ({ navigation, route
 
     bundleResolver.resolveAllBundles(params).then((bundle) => {
+      // Use cached schema name as fallback if no specific overlay name is found
+      const cachedSchemaName = getSchemaName(credential)
```
In the spirit of keeping your code [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself) should these three lines be moved into a helper function like `effectiveNameForCredential()`, similar to `getSchemaName` in cred-def.ts?

PatStLouis replied:
I think so, just had a chat with Bryce and we came to the same conclusion. I will rebase with the webvh pr that was merged and clean this up

PatStLouis replied:
I've implemented an getEffectiveCredentialName function

jleach commented on packages/core/src/screens/CredentialOfferAccept.tsx:
```diff
@@ -84,8 +86,78 @@ const CredentialOfferAccept: React.FC<CredentialOfferAcceptProps> = ({ visible,
+                  // Store schema metadata
+                  const metadataToStore = {
+                    schemaId: offerData.schema_id,
+                    credentialDefinitionId: 'cred_def_id' in offerData ? offerData.cred_def_id : undefined,
```
Is this hard coded property `cred_def_id` specific to `webhv` or a test credential you're working with. Another way, will every webhv credential have this propert?

PatStLouis replied:
This comes from credo and is already a default value, not webvh sepcific:
https://github.com/openwallet-foundation/bifold-wallet/blob/165cb5388d2ec57ecdfd4baf0812cc5317101c3f/packages/core/src/screens/CredentialOffer.tsx#L130

PatStLouis committed:
refactor: centralize credential metadata caching and naming logic

Major refactoring to eliminate code duplication and improve credential naming:

### New Utilities Added:

**credential.ts:**
- ensureCredentialMetadata(): Centralized function to cache schemaName and credDefTag
- getEffectiveCredentialName(): Priority waterfall for credential display names

**cred-def.ts:**
- getSchemaName(): Extract cached schema name from metadata
- getCredDefTag(): Extract cached cred def tag from metadata
- fallbackDefaultCredentialNameValue: Global constant for default name
- defaultCredDefTag: Global constant for default tag value

### Priority Waterfall (Consistent Everywhere):
1. OCA Bundle name (if meaningful)
2. Credential definition tag (skip if 'default')
3. Schema name (from cached metadata)
4. Fallback to 'Credential'

### Refactored Files:

**CredentialOffer.tsx:**
- Replaced ~40 lines of resolution logic with ensureCredentialMetadata()
- Uses getEffectiveCredentialName() for history logging

**CredentialOfferAccept.tsx:**
- Replaced ~65 lines with ensureCredentialMetadata()
- Removed redundant metadata checking

**CredentialDetails.tsx:**
- Replaced ~30 lines of diagnostic logging with ensureCredentialMetadata()
- Uses getEffectiveCredentialName() for display and history

**CredentialCard10/11.tsx, CredentialDetailPrimaryHeader.tsx:**
- Simplified overlay name fallback logic from ~20 lines to 1-2 lines
- Single setOverlay call instead of two

**helpers.ts:**
- Updated credNameFromRestriction() to follow priority waterfall
- Proof request processing now uses cached metadata when available
- Replaced hardcoded 'default' and 'credential' with constants

### Metadata Now Cached:
- schemaId
- credentialDefinitionId
- schemaName
- credDefTag (NEW)

### Benefits:
- ~200+ lines of duplicated code eliminated
- Single source of truth for credential naming
- Faster (uses cached metadata instead of repeated async calls)
- More maintainable and testable
- Consistent naming across all screens and components
- No breaking changes to existing exports

Signed-off-by: Patrick St-Louis <patrick.st-louis@opsecid.ca>

PatStLouis committed:
Merge branch 'main' into feature/anoncreds-metadata-caching

PatStLouis committed:
chore: revert formatting changes to cred-def.test.ts

Remove unintentional formatting changes from test file.

Signed-off-by: Patrick St-Louis <patrick.st-louis@opsecid.ca>

PatStLouis committed:
refactor: enhance parsedCredDefNameFromCredential with metadata caching

Update parsedCredDefNameFromCredential to automatically ensure metadata is cached when
an agent is provided, then use the full priority waterfall logic via getEffectiveCredentialName.

Changes:
- Call ensureCredentialMetadata if agent is provided
- Use getEffectiveCredentialName for cached credentials (priority waterfall logic)
- Maintain fallback to ID parsing for backward compatibility with older credentials

This ensures consistent naming behavior across the codebase while maintaining compatibility
with existing credentials that don't have cached metadata.

Signed-off-by: Patrick St-Louis <patrick.st-louis@opsecid.ca>

PatStLouis committed:
fix: remove unused imports and update test snapshots

Remove unused imports:
- AnonCredsCredentialMetadataKey from CredentialOffer.tsx
- getCredentialName from helpers.ts

Update W3cProofRequest test snapshot to reflect new credential naming behavior.
The credNameFromRestriction function now uses the priority waterfall logic,
so credentials without cached metadata use the fallback name 'Credential'
instead of 'latest'.

All tests passing: 447 passed, 2 skipped

Signed-off-by: Patrick St-Louis <patrick.st-louis@opsecid.ca>

PatStLouis committed:
test: add comprehensive tests for credential naming and metadata caching

Add 6 new tests covering the new refactored functionality:

getEffectiveCredentialName tests:
- Verifies OCA name priority (highest)
- Verifies credDefTag fallback when OCA not available
- Verifies 'default' credDefTag is skipped in favor of schemaName
- Verifies fallback to 'Credential' when no metadata

ensureCredentialMetadata tests:
- Verifies schemaName and credDefTag are cached when missing
- Verifies existing cached metadata is not overwritten

All tests passing: 18/18 (12 original + 6 new)

Signed-off-by: Patrick St-Louis <patrick.st-louis@opsecid.ca>

PatStLouis committed:
refactor: simplify credential name handling and add comprehensive tests

- Refactor getEffectiveCredentialName with isValidName helper for better maintainability
- Remove redundant conditional checks in CredentialCard10, CredentialCard11, and CredentialDetails
- Simplify metaOverlay.name assignment by leveraging guaranteed fallback behavior
- Add comprehensive unit tests for getEffectiveCredentialName function
- Add integration tests for credential name handling in helpers context
- Fix CredentialCard11 to properly handle optional credential and metaOverlay props
- All 13 new tests passing with no linting errors

This refactoring reduces code duplication and improves readability while maintaining
the same functionality and proper fallback behavior.

Signed-off-by: Patrick St-Louis <patrick.st-louis@opsecid.ca>

bryce-mcmath commented on .changeset/add-anoncreds-metadata-caching.md:
Can we please make this a patch changeset instead of a minor one? We don't follow semantic versioning as new features and breaking changes happen almost every week so the version would balloon to a crazy high number so quickly.

PatStLouis replied:
I just made this into a patch

bryce-mcmath commented on packages/core/__tests__/screens/__snapshots__/W3cProofRequest.test.tsx.snap:
```diff
@@ -1044,13 +1044,13 @@ exports[`displays a proof request screen ProofRequest Screen, W3C displays a pro
                               Array [
                                 Object {
-                                  "backgroundColor": "#5f0f7b",
+                                  "backgroundColor": "#781e4b",
```
Q: What is causing this background colour to change?

PatStLouis replied:
> 7 Months Ago (April 2025):
> Commit b2987d7b: Filipe updated the OCA Branding code
> The actual running code changed to use #781e4b
> BUT the test snapshot file wasn't fully updated - it still had some old #5f0f7b values

> Yesterday (Your Commit ec60e0d0):
> You made changes to credential naming logic
> You ran the tests
> Jest noticed: "Hey, the snapshot says #5f0f7b but the actual output is #781e4b"
> Jest updated the snapshot to match what the code actually produces

PatStLouis replied:
does this make sense to you?

bryce-mcmath replied:
Not really just because snapshot changes are detected whenever the tests run and tests run as part of CICD so it would definitely be picked up and fail CICD in the last seven months.

Thinking more about it, I'm guessing the credential name is used to generate a random hex colour, and since it changed, so did the colour.

bryce-mcmath replied:
Yes, the random colour is generated from a hash of the credential definition ID, so it makes sense that it would change since the cred def tag was changed. All good

bryce-mcmath commented on packages/core/src/components/views/CredentialDetailPrimaryHeader.tsx:
```diff
@@ -30,6 +31,10 @@ const CredentialDetailPrimaryHeader: React.FC<CredentialDetailPrimaryHeaderProps
   const { width, height } = useWindowDimensions()
   const isBranding11 = brandingOverlayType === BrandingOverlayType.Branding11
+
+  // Use cached schema name as fallback if no specific overlay name is found
```
nit: I don't think this comment is needed here, probably suffices to have a similar comment in the actual getEffectiveCredentialName function itself

PatStLouis replied:
removed

bryce-mcmath commented on packages/core/src/screens/CredentialDetails.tsx:
```diff
@@ -176,10 +176,27 @@ const CredentialDetails: React.FC<CredentialDetailsProps> = ({ navigation, route
+    }
+
+    const restoreMetadata = async () => {
+      await ensureCredentialMetadata(credential, agent)
```
If it's possible for this async function  to throw an error, it should be wrapped in a try catch block

PatStLouis replied:
done

bryce-mcmath commented on packages/core/src/screens/CredentialOffer.tsx:
```diff
@@ -128,9 +126,10 @@ const CredentialOffer: React.FC<CredentialOfferProps> = ({ navigation, credentia
-        credential.metadata.add(AnonCredsCredentialMetadataKey, {
-          schemaId: offerData.schema_id,
-          credentialDefinitionId: offerData.cred_def_id,
+        // Ensure credential has all required metadata
```
nit: This comment is kind of a restatement of the function name and is not needed imo

PatStLouis replied:
comment removed

bryce-mcmath commented on packages/core/src/screens/CredentialOffer.tsx:
```diff
@@ -128,9 +126,10 @@ const CredentialOffer: React.FC<CredentialOfferProps> = ({ navigation, credentia
+        await ensureCredentialMetadata(credential, agent, {
+          schema_id: offerData.schema_id,
+          cred_def_id: offerData.cred_def_id,
         })
```
If this awaited function can throw, it should probably be wrapped in a try catch block

PatStLouis replied:
done

bryce-mcmath commented on packages/core/src/screens/CredentialOfferAccept.tsx:
```diff
@@ -84,8 +86,16 @@ const CredentialOfferAccept: React.FC<CredentialOfferAcceptProps> = ({ visible,
       timer && clearTimeout(timer)
       setCredentialDeliveryStatus(DeliveryStatus.Completed)
+
+      // Ensure credential has all required metadata (function checks what's missing)
```
nit: same thing as above, comment can be removed

PatStLouis replied:
comment removed

bryce-mcmath commented on packages/core/src/screens/CredentialOfferAccept.tsx:
```diff
@@ -84,8 +86,16 @@ const CredentialOfferAccept: React.FC<CredentialOfferAcceptProps> = ({ visible,
+      // Ensure credential has all required metadata (function checks what's missing)
+      const restoreMetadata = async () => {
+        if (agent) {
+          await ensureCredentialMetadata(credential, agent)
```
Same thing as above, wrap in try catch if it can throw

PatStLouis replied:
done

bryce-mcmath commented on packages/core/src/utils/cred-def.ts:
```diff
@@ -86,8 +93,35 @@ function credentialDefinition(credential: CredentialRecord): string | undefined
-  return getCredentialName(credentialDefinition(credential), credentialSchema(credential), agent)
+  // Ensure metadata is cached if agent is provided
+  if (agent) {
+    await ensureCredentialMetadata(credential, agent)
```
Same as above; try catch

PatStLouis replied:
done

bryce-mcmath commented on packages/core/src/utils/credential.ts:
```diff
@@ -35,3 +37,143 @@ export const getCredentialIdentifiers = (credential: CredentialExchangeRecord) =
+      const formatOfferData = offer?.anoncreds ?? offer?.indy
+      if (formatOfferData && typeof formatOfferData === 'object') {
+        schemaId = schemaId || ('schema_id' in formatOfferData ? formatOfferData.schema_id : undefined)
+        credDefId = credDefId || ('cred_def_id' in formatOfferData ? formatOfferData.cred_def_id : undefined)
```
Would this be simpler?

```suggestion
        schemaId = schemaId || formatOfferData?.schema_id
        credDefId = credDefId || formatOfferData?.cred_def_id
```

PatStLouis replied:
had to do slightly different:
```
schemaId = schemaId || (formatOfferData as any).schema_id
credDefId = credDefId || (formatOfferData as any).cred_def_id
```

bryce-mcmath replied:
`as any` should be avoided as much as possible

if it's possible for formatData to be undefined or null or a non-object then we should still use optional chaining. If you still have to cast with `as` try to use the correct type, not `any`

PatStLouis replied:
I imported the AnonCredsCredentialOffer type

bryce-mcmath commented on packages/core/src/utils/credential.ts:
```diff
@@ -35,3 +37,143 @@ export const getCredentialIdentifiers = (credential: CredentialExchangeRecord) =
+      name !== fallbackDefaultCredentialNameValue &&
+      name.trim() !== ''
+    )
+  }
```
This function can be moved out of the parent function so it doesn't need to be redeclared every time the parent gets called. Would also make it easier to test it independently

PatStLouis replied:
done, test added

bryce-mcmath reviewed (changes requested):
Small nits, good work so far

PatStLouis committed:
feat: improve credential metadata error handling and refactor validation

- Add try-catch blocks around ensureCredentialMetadata calls
- Gracefully handle metadata restoration failures with fallback to default names
- Extract isValidCredentialName as standalone exported function
- Add comprehensive tests for isValidCredentialName (8 new test cases)
- Fix formatOfferData property access with proper type assertion
- Add error handling in CredentialDetails, CredentialOfferAccept, and cred-def
- All error cases now log warnings instead of crashing the app

Error handling improvements ensure app continues to function even when:
- Network requests fail
- Schema/CredDef resolution fails
- Metadata caching encounters errors

All 15 tests passing with no linting errors.

Signed-off-by: Patrick St-Louis <patrick.st-louis@opsecid.ca>

PatStLouis committed:
fix: replace console.warn with proper logger calls and remove unused imports

- Remove unused fallbackDefaultCredentialNameValue imports
- Replace console.warn with agent.config.logger?.warn for proper logging
- Remove unused comment in CredentialDetailPrimaryHeader
- Fix linting errors reported by CI/CD

All tests passing (473 passed, 2 skipped)

Signed-off-by: Patrick St-Louis <patrick.st-louis@opsecid.ca>

PatStLouis committed:
chore: change anoncreds-metadata-caching to patch version bump

Change from minor to patch since these are bug fixes and improvements
rather than new features.

Signed-off-by: Patrick St-Louis <patrick.st-louis@opsecid.ca>

PatStLouis committed:
fix: add proper type annotation for AnonCreds credential offer

- Import AnonCredsCredentialOffer type from @credo-ts/anoncreds
- Type formatOfferData as AnonCredsCredentialOffer | undefined
- Remove need for type assertions and guards
- Both anoncreds and indy formats use the same AnonCredsCredentialOffer type
- Improve type safety and code readability

All tests passing (15/15 credential tests)

Signed-off-by: Patrick St-Louis <patrick.st-louis@opsecid.ca>

jleach commented on packages/core/src/screens/CredentialDetails.tsx:
```diff
@@ -176,10 +176,32 @@ const CredentialDetails: React.FC<CredentialDetailsProps> = ({ navigation, route
+        await ensureCredentialMetadata(credential, agent)
+      } catch (error) {
+        // If metadata restoration fails, we'll fall back to default credential name
+        agent?.config.logger?.warn('Failed to restore credential metadata', { error: error as Error })
```
By convention we either pass in logger or get it from dependency injection.

```TypeScript
  const [logger] = useServices([TOKENS.UTIL_LOGGER])
```

jleach commented on packages/core/src/screens/CredentialOfferAccept.tsx:
```diff
@@ -84,8 +86,19 @@ const CredentialOfferAccept: React.FC<CredentialOfferAcceptProps> = ({ visible,
+          try {
+            await ensureCredentialMetadata(credential, agent)
+          } catch (error) {
+            agent.config.logger?.warn('Failed to restore credential metadata', { error: error as Error })
```
`logger.warn()` no?

jleach commented on packages/core/src/utils/cred-def.ts:
```diff
@@ -31,7 +38,7 @@ export async function getCredentialName(credDefId?: string, schemaId?: string, a
 }
 
 async function parseWebVHCredDefId(credDefId?: string, schemaId?: string, agent?: Agent): Promise<string> {
```
Maybe this fn just needed to pass in logger rather than agent?

jleach commented on packages/core/src/utils/cred-def.ts:
```diff
@@ -86,8 +93,40 @@ function credentialDefinition(credential: CredentialRecord): string | undefined
+      await ensureCredentialMetadata(credential, agent)
+    } catch (error) {
+      // If metadata restoration fails, we'll fall back to parsing IDs or default name
+      agent?.config.logger?.warn('Failed to restore credential metadata in parsedCredDefNameFromCredential', { error: error as Error })
```
I think passing in logger would be better. You can only write log messages if agent, which is optional, is passed in. Also, if this is a `catch` maybe you mean .error() rather than .warn()?

PatStLouis replied:
I passed in the logger, but the agent is still needed

PatStLouis replied:
I've updated the code, the agent isn't needed anymore. I changed the webvh tests to use the new resolve function.

jleach commented on packages/core/src/utils/credential.ts:
```diff
@@ -35,3 +37,149 @@ export const getCredentialIdentifiers = (credential: CredentialExchangeRecord) =
+        credDefId = credDefId || formatOfferData?.cred_def_id
+      }
+    } catch (error) {
+      agent.config.logger?.warn('Failed to get format data', { error: error as Error })
```
`logger.error()`? This seems like an error condition.

jleach commented on packages/core/src/utils/helpers.ts:
```diff
@@ -338,11 +339,46 @@ const credNameFromRestriction = async (
   })
-  if (schema_name && (schema_name.toLowerCase() !== 'default' || schema_name.toLowerCase() !== 'credential')) {
+
+  // Priority 1: Provided restriction schema name (from schema_name in restriction if meaningful)
```
Not critical for this PR but this fn is getting a bit long and hard to test. Writing 3 smaller functions then this one to just wrap 'em and return might make it easier to test:
```
getRestrictionSchemaName()
resolveCredentialDefenitionTagAsName()
resolveSchemaName()
```

PatStLouis replied:
I created 2 functions for the resolving operations. Since the schema name if just 1 conditional check, I say its fine to leave it

PatStLouis replied:
I updated the webvh tests to use these 2 new functions

jleach reviewed (changes requested):
- Some time ago Clecio asked us to rely less on `agent?.config.logger?` and use he one from dependency injection or pass it in to the function as an optional param.
- There is one really long fn. When they get overly long they're harder to test and maintain. Consider breaking it down into a few smaller functions and the main one just wraps them. Ref: https://en.wikipedia.org/wiki/Single-responsibility_principle
- Sonar noted that your imports need to be cleaned up. There are quite a few multiple imports:
['../../utils/credential' imported multiple times.](https://sonarcloud.io/project/issues?pullRequest=1681&open=AZoDndi7yhGybQZXtr-5&id=openwallet-foundation_bifold-wallet)

PatStLouis committed:
Merge branch 'openwallet-foundation:main' into feature/anoncreds-metadata-caching

PatStLouis committed:
fix: use BifoldLogger instance

Signed-off-by: Patrick St-Louis <patrick.st-louis@opsecid.ca>

PatStLouis committed:
fix: update missed helpers file

Signed-off-by: Patrick St-Louis <patrick.st-louis@opsecid.ca>

PatStLouis committed:
tests: Updated webvh tests to use new functions

Signed-off-by: Patrick St-Louis <patrick.st-louis@opsecid.ca>

PatStLouis committed:
fix: remove unused imports

Signed-off-by: Patrick St-Louis <patrick.st-louis@opsecid.ca>
